python/cuml/dask/preprocessing/encoders.py

```diff
@@ -0,0 +1,357 @@
+    The features are encoded using a one-hot (aka 'one-of-K' or 'dummy')
+    encoding scheme. This creates a binary column for each category and
+    returns a sparse matrix or dense array (depending on the ``sparse``
+    parameter).
+    By default, the encoder derives the categories based on the unique values
+    in each feature. Alternatively, you can also specify the `categories`
+    manually.
+
+    Parameters
+    ----------
+    categories : 'auto' or a dask_cudf.DataFrame, default='auto'
+        Categories (unique values) per feature:
+        - 'auto' : Determine categories automatically from the training data.
+        - DataFrame : ``categories[col]`` holds the categories expected in the
+          feature col.
+    drop : 'first', None or a dict, default=None
+        Specifies a methodology to use to drop one of the categories per
+        feature. This is useful in situations where perfectly collinear
+        features cause problems, such as when feeding the resulting data
+        into a neural network or an unregularized regression.
+        - None : retain all features (the default).
+        - 'first' : drop the first category in each feature. If only one
+          category is present, the feature will be dropped entirely.
+        - Dict : ``drop[col]`` is the category in feature col that
+          should be dropped.
+    sparse : bool, default=False
+        This feature was deactivated and will give an exception when True.
+        The reason is because sparse matrix are not fully supported by cupy
+        yet, causing incorrect values when computing one hot encodings.
+        See https://github.com/cupy/cupy/issues/3223
+    dtype : number type, default=np.float
+        Desired datatype of transform's output.
+    handle_unknown : {'error', 'ignore'}, default='error'
+        Whether to raise an error or ignore if an unknown categorical feature
+        is present during transform (default is to raise). When this parameter
+        is set to 'ignore' and an unknown category is encountered during
+        transform, the resulting one-hot encoded columns for this feature
+        will be all zeros. In the inverse transform, an unknown category
+        will be denoted as None.
+
+    Attributes
+    ----------
+    drop_idx_ : array of shape (n_features,)
+        ``drop_idx_[i]`` is the index in ``categories_[i]`` of the category to
+        be dropped for each feature. None if all the transformed features will
+        be retained.
+    """
+
+    def __init__(self, categories='auto', drop=None, sparse=False,
+                 dtype=np.float, handle_unknown='error', client=None):
+        self.categories = categories
+        self.sparse = sparse
+        self.dtype = dtype
+        self.handle_unknown = handle_unknown
+        self.drop = drop
+        self._fitted = False
+        self.drop_idx_ = None
+        self._encoders = None
+        self.client_ = client if client is not None else default_client()
```

Comment:
There's a new function in `cuml.dask.common.utils` called `get_client()`. I'd like to start seein that be used in place of doing this everywhere.

Reply:
Good Point!

Reply:
Good point.